### PR TITLE
Implement seed parameter and parameters testing

### PR DESCRIPTION
**Pull request by innovimax-jim**

### CODE REVIEW

File: src/main/java/innovimax/mixthem/arguments/Arguments.java [hunk 1]
```diff
@@ -142,7 +153,7 @@ public static void printUsage() {
         for(Rule rule : Rule.values()) {
           System.out.print("  - " + rule.getName());
           for(RuleParam param : rule.getParams()) {
-              System.out.print("#"+param.getName());
+              System.out.print(" [#"+param.getName()+"]");
```
Comment (innovimax) [inline src/main/java/innovimax/mixthem/arguments/Arguments.java]: in fact here, you should add "[" .. "]” only if param is not required

File: src/main/java/innovimax/mixthem/io/DefaultLineReader.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 	private final Path path;
 	private final BufferedReader reader;
 	private final boolean first;
-	private final Random random;
+	private Random random;
```
Comment (innovimax) [inline src/main/java/innovimax/mixthem/io/DefaultLineReader.java]: please make it final again see below why

File: src/main/java/innovimax/mixthem/io/DefaultLineReader.java [hunk 3]
```diff
@@ -42,6 +42,11 @@ public DefaultLineReader(File input, boolean first) throws IOException {
 	}
 
 	@Override
+	public void initSeed(int seed) {
+		this.random = new Random(seed);
```
Comment (innovimax) [inline src/main/java/innovimax/mixthem/io/DefaultLineReader.java]: you don't need to recreate a Random : just use setSeed on the existing Random

File: src/main/java/innovimax/mixthem/arguments/Arguments.java (hunk 1 above)
Comment (innovimax-jim) [inline src/main/java/innovimax/mixthem/arguments/Arguments.java]: it's true

File: src/main/java/innovimax/mixthem/io/DefaultLineReader.java (hunk 3 above)
Comment (innovimax-jim) [inline src/main/java/innovimax/mixthem/io/DefaultLineReader.java]: you're right

**Comment by innovimax-jim** [comment]
work on #79

**Comment by innovimax-jim** [comment]
need to implement -join #col1 #col2

**Comment by innovimax-jim** [comment]
@innovimax can you check it

**Pull request merged**